examples/applications/plot_cyclical_feature_engineering.py

```diff
@@ -0,0 +1,693 @@
+#   discontinuities at midnight but the linear regression model still fails to
+#   leverage those features to properly model intra-day variations. Using
+#   trigonmetric features for higher harmonics or additional trigonometric
+#   features for the natural period with different phases could potentially fix
+#   this problem.
+#
+# - the periodic spline-based features fix those two problems at once: they
+#   give more expressivity to the linear model by making it possible to focus
+#   on a specific hours thanks to the use of 12 knots. Furthermore the
+#   `extrapolation="periodic"` option enforces a smooth representation between
+#   `hour=23` and `hour=0`.
+#
+# - the one-hot encoded features behave similarly to the periodic spline-based
+#   features but are more spiky: for instance they can better model the morning
+#   peak during the weak days since this peak last fewer than an hour. However
+#   we will see in the following that what can be an advantage for linear
+#   models is not necessarily one for more expressive models.
+#
+# Finally we observe that none of the linear models can approximate the true
+# bike rentals demand, especially for the peaks that can be very sharp at rush
+# hours during the working days but much flatter during the week-ends: the most
+# accurate linear models based on splines or one-hot encoding tend to forecast
+# peaks of commuting-related bike rentals even on the week-ends and
+# under-estimate the commuting-related events during the working days.
+#
+# These systematic prediction errors reveal a form of under-fitting and can be
+# explained by the lack of non-additive modeling of the interactions between
+# features (in this case "workingday" and features derived from "hours"). This
+# issue will be addressed in the following section.
+#
+# Modeling non-linear feature interactions with kernels
+# -----------------------------------------------------
+#
+# The previous analysis highlighted the need to model the interactions between
+# "workingday" and "hours". Another example of a such a non-linear interactions
+# that we would like to model could be the impact of the rain that might not be
+# the same during the working days and the week-ends and holidays for instance.
+#
+# Linear models cannot capture non-linear interactions between features even if
+# the features them-selves are marginally non-linear as is the case with
+# features constructed by `SplineTransformer` (or one-hot encoding or binning).
+#
+# To capture such interactions we could either use a partial polynomial
+# expansion `PolynomicalFeatures(degree=2, interaction_only=True)` or use the
+# Nyström method to compute an approximate polynomial kernel expansion.
+#
+# Let us try the latter for computational reasons:
+from sklearn.preprocessing import MinMaxScaler
+from sklearn.kernel_approximation import Nystroem
+
+
+cyclic_spline_poly_pipeline = make_pipeline(
+    cyclic_spline_transformer,
+    MinMaxScaler(),
```

Comment:
```suggestion
```
I think this is not necessary. B-Splines are always in the range [0, 1]. The sum of all splines in one single point is always 1.

Reply:
Not all the features are pre-processed by splines (e.g. temperatures that are on a different scale). Probably they should and the performance could be better but that will make the story significantly more complex if we also use splines for the non time-related features.

Reply:
I see. `cyclic_spline_transformer` already processes all the columns. In that case, either leave it as is, or - just a suggestion - why not using `MinMaxScaler` for the "passthrough" features in `cyclic_spline_transformer`?

Reply:
Done.